File: frontend/src/types/benchmark.ts
/**
 * Multi-baseline benchmarking types (Figure 1).
 */

export interface BenchmarkBaseline {
  name: string;
  status?: string;
  wall_time_seconds: number | null;
  relative_speed: number | null;
  data_read_pct: number | null;
  atoms_analyzed_pct: number | null;
  index_bytes_mb: number | null;
}

export interface BenchmarkResponse {
  query_id: string;
  speedup_vs_mdanalysis: number;
  io_reduction_pct: number;
  data_read_pct: number;
  atoms_analyzed_pct: number;
  index_bytes_mb: number;
  baselines: BenchmarkBaseline[];
}

File: backend/app/schemas/benchmark.py
"""MOBench benchmarking schemas."""

from pydantic import BaseModel, Field
from typing import List, Optional

class BenchmarkBaseline(BaseModel):
    name: str
    status: str = "MEASURED"
    wall_time_seconds: Optional[float] = None
    relative_speed: Optional[float] = None
    data_read_pct: Optional[float] = None
    atoms_analyzed_pct: Optional[float] = None
    index_bytes_mb: Optional[float] = None

class BenchmarkResponse(BaseModel):
    query_id: str
    speedup_vs_mdanalysis: float
    io_reduction_pct: float
    data_read_pct: float
    atoms_analyzed_pct: float
    index_bytes_mb: float
    baselines: List[BenchmarkBaseline]
